### PR TITLE
Test `io` from `stdlib` in the CI

### PR DESCRIPTION
**Pull request by Thirumalai-Shaktivel**

### CODE REVIEW

**Comment by Thirumalai-Shaktivel** [comment]
@czgdp1807 Can you please provide me push access to your stdlib repo: https://github.com/czgdp1807/stdlib?

**Comment by czgdp1807** [comment]
You are already added as a collaborator.

**Comment by Thirumalai-Shaktivel** [comment]
Cool! Now, I understand the issue is not related to the allocatable initialization, but it appears after calling the `collect_npy` function.

**Comment by Thirumalai-Shaktivel** [comment]
I added many workarounds and commented out some tests. I will fix them in the subsequent PRs.
This PR is ready to merge.

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -656,6 +656,11 @@ jobs:
             ./build_test_lf.sh
             ./build_test_gf.sh
 
+            git checkout lf20
+            git checkout 1dc82c3fefc530fe047056758d83fb3290956e16
+            ./build_test_gf.sh
+            ./build_test_lf.sh
+
```
Comment (certik) [inline .github/workflows/CI.yml:663]: Why do we need to test both lf19 and lf20?

I think testing just one branch (the latest one) should be enough.

Comment (Thirumalai-Shaktivel) [reply .github/workflows/CI.yml:663]: I created a new branch to work independently, it only tests the `io` module. 
Yes, we have to just use one branch to test. I will merge my changes into lf19 and push.

Comment (Thirumalai-Shaktivel) [reply .github/workflows/CI.yml:663]: @certik How about fixing all the bug, make all the tests pass and then merge into lf19 branch?

Comment (certik) [reply .github/workflows/CI.yml:663]: What is preventing to make any workarounds you did in the IO module and merge them with the lf19 branch?

It's hard for me to keep track of all the different branches. The work is only done when we have one branch, and compile it (with some workarounds).

In the worst case we can temporarily have two branches, in which case document in CI.yml which one is which and create an issue with a plan (and a timeline) when the other branch will be merged. Let me know the current issues, and then let's decide.

Comment (czgdp1807) [reply .github/workflows/CI.yml:663]: Let's not wait for fixing the bugs. Our main target is to make progress. We can fix bugs later. Merge all your changes into `lf19` branch and then that should be okay? If there are any blockers/issues with merging your branch into `lf19` branch then please feel free share here and we can figure out a solution.

Comment (certik) [reply .github/workflows/CI.yml:663]: Yes, I want to figure out any solution that allows us to maintain just one branch and that is our state of the progress. Bugs have to be workarounded.

Comment (Thirumalai-Shaktivel) [reply .github/workflows/CI.yml:663]: Cool, I got it!
Now, after merging two tests in strings modules fail, I will look into it tomorrow, fix and report back.

**Comment by Thirumalai-Shaktivel** [comment]
Ready!

**Review by czgdp1807 — APPROVED** [review]

**Comment by Thirumalai-Shaktivel** [comment]
Cool, thanks for the review!

**Pull request merged**